plugins/out_oci_logan/oci_logan.c

```diff
@@ -0,0 +1,1327 @@
+                                   FLB_HASH_SHA256,
+                                   (unsigned char *) ctx->private_key,
+                                   flb_sds_len(ctx->private_key),
+                                   sha256_buf, sizeof(sha256_buf),
+                                   sig, &sig_len);
+
+
+    if (ret != FLB_CRYPTO_SUCCESS) {
+        flb_plg_error(ctx->ins, "error signing SHA256");
+        return NULL;
+    }
+
+    signature = flb_sds_create_size(512);
+    if (!signature) {
+        flb_errno();
+        return NULL;
+    }
+
+    /* base 64 encode */
+    len = flb_sds_alloc(signature) - 1;
+    flb_base64_encode((unsigned char*) signature, len, &outlen, sig,
+                          sizeof(sig));
+    signature[outlen] = '\0';
+    flb_sds_len_set(signature, outlen);
+
+    return signature;
+}
+
+static flb_sds_t get_date(void)
+{
+
+    flb_sds_t rfc1123date;
+    time_t t;
+    size_t size;
+    struct tm tm = { 0 };
+
+    /* Format Date */
+    rfc1123date = flb_sds_create_size(32);
+    if (!rfc1123date) {
+        flb_errno();
+        return NULL;
+    }
+
+    t = time(NULL);
+    if (!gmtime_r(&t, &tm)) {
+        flb_errno();
+        flb_sds_destroy(rfc1123date);
+        return NULL;
+    }
+    size = strftime(rfc1123date, flb_sds_alloc(rfc1123date) - 1,
+                    "%a, %d %b %Y %H:%M:%S GMT", &tm);
+    if (size <= 0) {
+        flb_errno();
+        flb_sds_destroy(rfc1123date);
+        return NULL;
+    }
+    flb_sds_len_set(rfc1123date, size);
+    return rfc1123date;
+}
+
+static flb_sds_t add_header_and_signing(struct flb_http_client *c,
+                                        flb_sds_t signing_str, const char *header, int headersize,
+                                        const char *val, int val_size)
+{
+    if (!signing_str) {
+        return NULL;
+    }
+
+    flb_http_add_header(c, header, headersize, val, val_size);
+
+    signing_str = flb_sds_cat(signing_str, "\n", 1);
+    signing_str = flb_sds_cat(signing_str, header, headersize);
+    signing_str = flb_sds_cat(signing_str, ": ", 2);
+    signing_str = flb_sds_cat(signing_str, val, val_size);
+
+    return signing_str;
+}
+
+static int build_headers(struct flb_http_client *c, struct flb_oci_logan *ctx,
+                         flb_sds_t json, flb_sds_t hostname, int port, flb_sds_t uri)
+{
+    int ret = -1;
+    flb_sds_t tmp_sds = NULL;
+    flb_sds_t signing_str = NULL;
+    flb_sds_t rfc1123date = NULL;
+    flb_sds_t encoded_uri = NULL;
+    flb_sds_t signature = NULL;
+    flb_sds_t auth_header_str = NULL;
+
+    flb_sds_t tmp_ref = NULL;
+
+    size_t tmp_len = 0;
+
+    unsigned char sha256_buf[32] = { 0 };
+
+    tmp_sds = flb_sds_create_size(512);
+    if (!tmp_sds) {
+        flb_errno();
+        goto error_label;
+    }
+
+    signing_str = flb_sds_create_size(1024);
+    if (!signing_str) {
+        flb_errno();
+        goto error_label;
+    }
+
+    // Add (requeset-target) to signing string
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#single-line-comment

Reply:
Fixed in all the places